src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs

```diff
@@ -688,6 +722,96 @@ private static string GetSearchText(string parameterString)
+            IntPtr selectionContainerPtr = IntPtr.Zero;
+            try
+            {
+                if (!ErrorHandler.Succeeded(VsMonitorSelection.GetCurrentSelection(out hierarchyPtr, out uint itemId, out IVsMultiItemSelect multiItemSelect, out selectionContainerPtr))
+                    || (itemId == VSConstants.VSITEMID_NIL && hierarchyPtr != IntPtr.Zero))
+                {
+                    return null;
+                }
+
+                var packages = new List<string>();
+                if (itemId != VSConstants.VSITEMID_SELECTION)
+                {
+                    // This is a single item selection.
+
+                    IVsHierarchy hierarchy = Marshal.GetTypedObjectForIUnknown(hierarchyPtr, typeof(IVsHierarchy)) as IVsHierarchy;
+                    if (hierarchy != null)
+                    {
+                        if (ErrorHandler.Succeeded(hierarchy.GetProperty(itemId, (int)__VSHPROPID7.VSHPROPID_ProjectTreeCapabilities, out object treeCapabilities)))
+                        {
+                            string packageName = ParsePackageNameFromTreeCapabilities(treeCapabilities?.ToString());
+                            if (packageName != null)
+                            {
+                                packages.Add(packageName);
+                            }
+                        }
+                    }
+                }
+                else if (multiItemSelect != null)
+                {
+                    // This is a multiple item selection.
+
+                    uint numberOfSelectedItems;
+                    int isSingleHierarchyInt;
+                    if (ErrorHandler.Succeeded(multiItemSelect.GetSelectionInfo(out numberOfSelectedItems, out isSingleHierarchyInt)))
+                    {
+                        bool isSingleHierarchy = isSingleHierarchyInt != 0;
+
+                        VSITEMSELECTION[] vsItemSelections = new VSITEMSELECTION[numberOfSelectedItems];
+                        uint flags = 0; // No flags, which will give us back a hierarchy for each item
+                        if (ErrorHandler.Succeeded(multiItemSelect.GetSelectedItems(flags, numberOfSelectedItems, vsItemSelections)) && isSingleHierarchy)
+                        {
+                            foreach (VSITEMSELECTION sel in vsItemSelections)
+                            {
+                                if (sel.pHier != null && ErrorHandler.Succeeded(sel.pHier.GetProperty(sel.itemid, (int)__VSHPROPID7.VSHPROPID_ProjectTreeCapabilities, out object treeCapabilities)))
+                                {
+                                    string packageName = ParsePackageNameFromTreeCapabilities(treeCapabilities?.ToString());
+                                    if (packageName != null)
+                                    {
+                                        packages.Add(packageName);
+                                    }
+                                }
```

Comment:
This code is similar to that above. Can you pass the hierarchy and item ID to `ParsePackageNameFromTreeCapabilities` to reduce duplication a little?